tests/ashley/test_models_lticontext.py

```diff
@@ -271,3 +271,17 @@ def test_get_user_roles_other_context(self):
             ["newgroup"],
             context2.get_user_roles(user),
         )
+
+    def test_get_course_info(self):
+        """
+        get_course_info() should extract the course informations from the
+        id of the context, when it is in the OpenEdX format.
+        """
+
+        context1 = LTIContextFactory(
+            lti_id="course-v1:myschool+mathematics101+session01"
+        )
+        course_info1 = context1.get_course_info()
+        self.assertEqual("myschool", course_info1["school_name"])
+        self.assertEqual("mathematics101", course_info1["course_name"])
+        self.assertEqual("session01", course_info1["course_run"])
```

Comment:
It might be good to also test non-openedx or partial patterns?

Reply:
Yes, good suggestion, thanks!